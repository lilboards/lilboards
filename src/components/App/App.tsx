import CssBaseline from '@mui/material/CssBaseline';
import { StrictMode } from 'react';
import { Provider } from 'react-redux';
import { RouterProvider } from 'react-router';
import {
  createBrowserRouter,
  createRoutesFromElements,
} from 'react-router-dom';
import routes from 'src/routes';
import { store } from 'src/store';

const router = createBrowserRouter(createRoutesFromElements(routes));

export default function App() {
  return (
    <StrictMode>
      <CssBaseline />
      <Provider store={store}>
        <RouterProvider router={router} />
      </Provider>
    </StrictMode>
  );
}
