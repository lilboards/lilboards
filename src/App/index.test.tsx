import { render, screen } from '@testing-library/react';
import App from '.';

it('renders correctly', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});

it('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
