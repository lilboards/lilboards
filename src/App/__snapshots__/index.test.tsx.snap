// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</div>
`;
